numpy/_core/tests/test_regression.py

```diff
@@ -458,23 +458,24 @@ def test_pickle_py2_bytes_encoding(self):
         test_data = [
             # (original, py2_pickle)
             (np.str_('\u6f2c'),
-             b"cnumpy.core.multiarray\nscalar\np0\n(cnumpy\ndtype\np1\n"
+             b"cnumpy._core.multiarray\nscalar\np0\n(cnumpy\ndtype\np1\n"
```

Comment:
This change makes me realize we might also need to work on NumPy's pickle code so old pickle files don't generate deprecation warnings that downstream users can do nothing about without recreating the pickle file.

Reply:
IMO all of these regression tests should pass untouched with the old `numpy.core` paths. Hopefully the pickle API allows NumPy to be smart enough to replace that path inline while loading the pickle.

Reply:
I remember testing it - if a pickle contains `numpy.core.*` due to the fact that it was pickled with 1.x, then unpickling with 2.x emits a warning. 

I will revert changes here and add filter warning for now.I propose I will work on warning-less unpickling for these python 2 formats in a follow-up PR, is it OK?

Reply:
> I will revert changes here and add filter warning for now.I propose I will work on warning-less unpickling for these python 2 formats in a follow-up PR, is it OK?

SGTM, I just opened https://github.com/numpy/numpy/issues/24844 to track this. 

It would be nice if we could check that it is actually feasible to handle this inside NumPy before merging this PR, but no need to block merging this until the fix is implemented. It may be the case that the imports happen inside of the pickle internals and there is no way to NumPy to intercept the import.